src/main/java/com/example/onlinebookstore/controller/BookController.java

```diff
@@ -25,30 +28,36 @@ public class BookController {
 
     @GetMapping
     @Operation(summary = "Get all books", description = "Get a list of available books")
-    public List<BookDto> getAll(Pageable pageable) {
+    @PreAuthorize("hasRole('ROLE_USER')")
+    public List<BookDto> getAll(@PageableDefault(size = 20, sort = "id",
+            direction = Sort.Direction.ASC) Pageable pageable) {
         return bookService.findAll(pageable);
     }
 
     @GetMapping("/{id}")
     @Operation(summary = "Get book by ID", description = "Get a book by ID")
+    @PreAuthorize("hasRole('ROLE_USER')")
     public BookDto getBookById(@PathVariable Long id) {
         return bookService.getBookById(id);
     }
 
     @PostMapping
     @Operation(summary = "Create book", description = "Create a new book")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
     public BookDto createBook(@RequestBody @Valid CreateBookRequestDto requestDto) {
         return bookService.save(requestDto);
     }
 
     @DeleteMapping("/{id}")
     @Operation(summary = "Delete book", description = "Delete a book by ID")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
     public void delete(@PathVariable Long id) {
```

Comment:
You can add validation for pathvariables, like `@Positive`, for this you can add annotation `@Validated` on class